src/Makefile

```diff
@@ -405,6 +412,11 @@ ifneq ($(strip $(PREV_FINAL_LDFLAGS)), $(strip $(FINAL_LDFLAGS)))
 .make-prerequisites: persist-settings
 endif
 
+ifneq ($(strip $(PREV_STD)), $(strip $(STD)))
+.make-prerequisites: persist-settings
+endif
+
+
 # redis-server
 $(SERVER_NAME): $(ENGINE_SERVER_OBJ)
```

Comment:
if we are going with symlinks, we don't need this flexibility. we should always generate our preferred name and then create a symlink to it, like redis-server --> valkey-server.